File: leetcode-04/solutions/cpp/0532.cpp
class Solution {
 public:
  int findPairs(vector<int>& nums, int k) {
    if (k < 0) return 0;

    int ans = 0;
    unordered_map<int, int> map;

    for (int i = 0; i < nums.size(); i++) map[nums[i]] = i;

    for (int i = 0; i < nums.size(); i++)
      if (map.count(nums[i] + k) && map[nums[i] + k] != i) {
        ans++;
        map.erase(nums[i] + k);
      }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0532.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        if k < 0:
            return 0

        ans = 0
        dict = {}

        for i in range(len(nums)):
            dict[nums[i]] = i

        for i in range(len(nums)):
            if nums[i] + k in dict and dict[nums[i] + k] != i:
                ans += 1
                del dict[nums[i] + k]

        return ans
